src/include/constants.h

```diff
@@ -26,150 +26,211 @@
-        or at the least not be able to report the new error correctly.
-     */
-
-    PRTE_ERR_RECV_LESS_THAN_POSTED = (PRTE_ERR_SPLIT - 1),
-    PRTE_ERR_RECV_MORE_THAN_POSTED = (PRTE_ERR_SPLIT - 2),
-    PRTE_ERR_NO_MATCH_YET = (PRTE_ERR_SPLIT - 3),
-    PRTE_ERR_REQUEST = (PRTE_ERR_SPLIT - 4),
-    PRTE_ERR_NO_CONNECTION_ALLOWED = (PRTE_ERR_SPLIT - 5),
-    PRTE_ERR_CONNECTION_REFUSED = (PRTE_ERR_SPLIT - 6),
-    PRTE_ERR_COMPARE_FAILURE = (PRTE_ERR_SPLIT - 9),
-    PRTE_ERR_COPY_FAILURE = (PRTE_ERR_SPLIT - 10),
-    PRTE_ERR_PROC_STATE_MISSING = (PRTE_ERR_SPLIT - 11),
-    PRTE_ERR_PROC_EXIT_STATUS_MISSING = (PRTE_ERR_SPLIT - 12),
-    PRTE_ERR_INDETERMINATE_STATE_INFO = (PRTE_ERR_SPLIT - 13),
-    PRTE_ERR_NODE_FULLY_USED = (PRTE_ERR_SPLIT - 14),
-    PRTE_ERR_INVALID_NUM_PROCS = (PRTE_ERR_SPLIT - 15),
-    PRTE_ERR_ADDRESSEE_UNKNOWN = (PRTE_ERR_SPLIT - 16),
-    PRTE_ERR_SYS_LIMITS_PIPES = (PRTE_ERR_SPLIT - 17),
-    PRTE_ERR_PIPE_SETUP_FAILURE = (PRTE_ERR_SPLIT - 18),
-    PRTE_ERR_SYS_LIMITS_CHILDREN = (PRTE_ERR_SPLIT - 19),
-    PRTE_ERR_FAILED_GET_TERM_ATTRS = (PRTE_ERR_SPLIT - 20),
-    PRTE_ERR_WDIR_NOT_FOUND = (PRTE_ERR_SPLIT - 21),
-    PRTE_ERR_EXE_NOT_FOUND = (PRTE_ERR_SPLIT - 22),
-    PRTE_ERR_PIPE_READ_FAILURE = (PRTE_ERR_SPLIT - 23),
-    PRTE_ERR_EXE_NOT_ACCESSIBLE = (PRTE_ERR_SPLIT - 24),
-    PRTE_ERR_FAILED_TO_START = (PRTE_ERR_SPLIT - 25),
-    PRTE_ERR_FILE_NOT_EXECUTABLE = (PRTE_ERR_SPLIT - 26),
-    PRTE_ERR_HNP_COULD_NOT_START = (PRTE_ERR_SPLIT - 27),
-    PRTE_ERR_SYS_LIMITS_SOCKETS = (PRTE_ERR_SPLIT - 28),
-    PRTE_ERR_SOCKET_NOT_AVAILABLE = (PRTE_ERR_SPLIT - 29),
-    PRTE_ERR_SYSTEM_WILL_BOOTSTRAP = (PRTE_ERR_SPLIT - 30),
-    PRTE_ERR_RESTART_LIMIT_EXCEEDED = (PRTE_ERR_SPLIT - 31),
-    PRTE_ERR_INVALID_NODE_RANK = (PRTE_ERR_SPLIT - 32),
-    PRTE_ERR_INVALID_LOCAL_RANK = (PRTE_ERR_SPLIT - 33),
-    PRTE_ERR_UNRECOVERABLE = (PRTE_ERR_SPLIT - 34),
-    PRTE_ERR_MEM_LIMIT_EXCEEDED = (PRTE_ERR_SPLIT - 35),
-    PRTE_ERR_HEARTBEAT_LOST = (PRTE_ERR_SPLIT - 36),
-    PRTE_ERR_PROC_STALLED = (PRTE_ERR_SPLIT - 37),
-    PRTE_ERR_NO_APP_SPECIFIED = (PRTE_ERR_SPLIT - 38),
-    PRTE_ERR_NO_EXE_SPECIFIED = (PRTE_ERR_SPLIT - 39),
-    PRTE_ERR_COMM_DISABLED = (PRTE_ERR_SPLIT - 40),
-    PRTE_ERR_FAILED_TO_MAP = (PRTE_ERR_SPLIT - 41),
-    PRTE_ERR_SENSOR_LIMIT_EXCEEDED = (PRTE_ERR_SPLIT - 42),
-    PRTE_ERR_ALLOCATION_PENDING = (PRTE_ERR_SPLIT - 43),
-    PRTE_ERR_NO_PATH_TO_TARGET = (PRTE_ERR_SPLIT - 44),
-    PRTE_ERR_OP_IN_PROGRESS = (PRTE_ERR_SPLIT - 45),
-    PRTE_ERR_OPEN_CONDUIT_FAIL = (PRTE_ERR_SPLIT - 46),
-    PRTE_ERR_DUPLICATE_MSG = (PRTE_ERR_SPLIT - 47),
-    PRTE_ERR_OUT_OF_ORDER_MSG = (PRTE_ERR_SPLIT - 48),
-    PRTE_ERR_FORCE_SELECT = (PRTE_ERR_SPLIT - 49),
-    PRTE_ERR_JOB_CANCELLED = (PRTE_ERR_SPLIT - 50),
-    PRTE_ERR_CONDUIT_SEND_FAIL = (PRTE_ERR_SPLIT - 51)
-};
-
-#define PRTE_ERR_MAX (PRTE_ERR_SPLIT - 100)
+#define PRTE_SUCCESS                               PMIX_SUCCESS
```

Comment:
Why not straight up removing PRTE_XYZ and use PMIX_XYZ? That would remove the need to this giant conversion table.

Reply:
I thought about that, and I agree. Would touch literally every file in prrte though. @rhc54 ?

Reply:
Ideally, that is what we would do - but it would take a fair amount of effort. You can't just do a blanket replace of `PRTE_ERR` with `PMIX_ERR` as we don't have 100% convergence - i.e., there are PRRTE errors that have no corresponding PMIx values.

I suppose we could fix that by defining some additional PMIx error constants, but that might feel a little too implementation-specific. On the other hand, it _is_ the "PMIx Reference RTE", so perhaps that justifies a little preferential treatment.

Thoughts?

Reply:
PRRTE (and this PR) already has PRRTE specific error codes that are not mapped in PMIx. We would just change the ones that are mapped.

If the PRRTE specific ones will never be used in PMIx, it might make sense to just leave them here.

Reply:
Understood - but how will a programmer know whether to expect a PMIx or a PRTE error code? Seems to me that you need to standardize on one or the other - otherwise, the code will become very hard to decipher.

Reply:
That seems to support the usage of all `PMIX_` codes.

Reply:
Or that we use the table so PRRTE always uses `PRTE_ERR_` codes, which is the way OMPI works with ORTE and OPAL - and ORTE works with OPAL for that matter.

I'm not advocating either direction - neither is perfect. Pushing a bunch of codes into PMIx raises the question of how general they are versus PRRTE/OMPI-specific. I suspect the translation table is the only viable option.

The concern I have right now is that we don't seem to be using a number of the PRRTE codes that we _used_ to return. Makes me suspicious that something got lost, likely in the ODLS as it reports error states that might lack sufficient atomicity to accurately report the cause of the failure.

Reply:
I'm fine with either way.

Regarding the removed error codes, I can put them back. But I suspected a lot of them are from long ago, seeing as they weren't used in the code. They can also always be put back as you see they are needed. I'm not opposed to either.